docs/installation_guide.md

```diff
@@ -17,10 +17,14 @@ Building OSM from source requires more steps but is the best way to test the lat
 You must have a working [Go](https://golang.org/doc/install) environment.
 
 ```console
-$ git clone git@github.com:openservicemesh/osm.git
+$ cd $GOPATH
+$ mkdir -p src/github.com/openservicemesh
+$ cd src/github.com/openservicemesh
```

Comment:
These shouldn't be required and seems rather environment specific.

For example in my env, `GOPATH=$HOME/go`, and the repo is cloned to `$HOME/src/osm`

Reply:
OK, I might have mixed up this requirement with other issues (wrong Go version)... I will re-test the installation from source and probably close this PR.

Reply:
With Go Modules, we no longer need to create projects inside the GOPATH, but I remember struggling a lot before Go 1.11 (: